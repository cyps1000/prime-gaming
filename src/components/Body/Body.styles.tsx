/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: "0",
    height: "300vh",
  },
}));

export { useStyles };
