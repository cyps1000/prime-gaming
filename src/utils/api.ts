import axios from "axios";

/**
 * Defines the hook props interface
 */
interface Config {
  mock: boolean;
}

export const getApiClient = (config: Config) => {
  const { mock } = config;

  /**
   * Handles getting the base api url
   */
  const getApiUrl = () => {
    if (process.env.NODE_ENV === "development") {
      if (mock) return "/";
      return process.env.REACT_APP_LOCAL_API;
    }
    return process.env.REACT_APP_STAGING_API;
  };

  const apiClient = axios.create({
    baseURL: getApiUrl(),
  });

  return { apiClient };
};
