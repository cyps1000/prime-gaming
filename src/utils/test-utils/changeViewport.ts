/**
 * External imports
 */
import mediaQuery from "css-mediaquery";

/**
 * Defines the BreakpointKey type
 */
type BreakpointKey = "xl" | "lg" | "md" | "sm" | "xs";

/**
 * Handles getting the width based on the breakpoint key provided
 */
const getWidthByThemeBreakpoints = (breakpointKey: BreakpointKey) => {
  switch (breakpointKey) {
    case "xl":
      return 1920;
    case "lg":
      return 1280;
    case "md":
      return 960;
    case "sm":
      return 600;
    case "xs":
      return 414;
    default:
      return 1920;
  }
};

/**
 * Handles changing the viewport
 */
export const changeViewport = (breakpointKey: BreakpointKey): any => {
  let width: number = getWidthByThemeBreakpoints(breakpointKey);

  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width: width }),
      addListener: () => {},
      removeListener: () => {},
    };
  };
};
