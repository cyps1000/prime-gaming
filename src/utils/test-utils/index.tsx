import React, { ReactElement } from "react";
/**
 * External Imports
 */
import { MemoryRouter } from "react-router-dom";

/**
 * Imports testing utils
 */
import { render, RenderOptions } from "@testing-library/react";

/**
 * Imports providers
 */

import { ThemeProvider, LanguageProvider } from "../../hooks";
import Providers from "../../components/Providers";

/**
 * Defines the Wrapper component
 */
const Wrapper: React.FC = ({ children }) => {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <Providers>
          <MemoryRouter>{children}</MemoryRouter>
        </Providers>
      </LanguageProvider>
    </ThemeProvider>
  );
};

/**
 * Defines the custom render
 */
const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

/**
 * Re-exports everything from the testing library
 */
export * from "@testing-library/react";

export * from "./changeViewport";
export * from "./mocks";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
