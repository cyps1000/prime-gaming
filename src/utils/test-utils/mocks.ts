export const mockHistoryPush = jest.fn();
