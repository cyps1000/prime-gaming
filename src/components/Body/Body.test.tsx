/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 * @see https://www.smashingmagazine.com/2020/07/react-apps-testing-library/
 */

/**
 * Imports test utils
 */
import { render, fireEvent } from "../../utils/test-utils";
import { mockHistoryPush } from "../../utils/test-utils/mocks";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import Body from "./Body";

/**
 * Body rendering tests
 */
describe("Body Rendering Tests", () => {
  it("renders the component without errors", () => {
    const { getByText } = render(<Body>Children</Body>);
    expect(getByText("Children")).toBeInTheDocument();
  });
});

/**
 * Body snapshot test
 */
describe("Body Snapshot Test", () => {
  it("passes the snapshot test", () => {
    const { container, getByText } = render(<Body>Children</Body>);
    expect(getByText("Children")).toBeInTheDocument();
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it("has a default className", () => {
    const { container, getByText } = render(<Body> Children </Body>);
    expect(container.firstChild).toHaveAttribute("class");
    expect(getByText("Children")).toBeInTheDocument();
  });
});
