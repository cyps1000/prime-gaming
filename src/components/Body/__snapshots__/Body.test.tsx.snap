// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Snapshot Test passes the snapshot test 1`] = `"<div class=\\"makeStyles-root-2\\">Children</div>"`;
